filebeat/beater/channels.go

```diff
@@ -67,6 +77,7 @@ func (c *publisherChannel) Close() { close(c.done) }
 func (c *publisherChannel) Send(events []*input.Event) bool {
 	select {
 	case <-c.done:
+		c.ch = nil
```

Comment:
Oh, I thought it's a common known golang pattern. assign nil to channel to block on channel forever.